[#57] - Implement endpoints for liking/unliking a post

LemonPepperSeasoning opened the pull request:
# Description:
Please provide a summary of changes made and the relevant issue that has been worked on. Also list any major changes made to project structure/dependencies if applicable

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](../CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My code follows the code style of this project
- [x] My code has been commented
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

LemonPepperSeasoning committed:
Like & unlike endpoint implemented

LemonPepperSeasoning committed:
Added a test and Fixed a small mistake

LemonPepperSeasoning committed:
fixed a bug in a unit test

Dj5822 commented on backend/__tests__/posts.js:
```diff
@@ -397,6 +452,64 @@ describe('Posts', () => {
             })
         })
 
+        describe('when user unlike post in a valid way', () => {
```
Move all the tests that you added into a separate describe() where you indicate the path of the request.

LemonPepperSeasoning replied:
Thanks for the review and the comment. 
I have just made the change !
Let me know if you think it is ready to be merged

Dj5822 reviewed (changes requested):
Looks good, except I think you should organize the tests into a separate describe() function. Currently it looks like the new tests are all inside describe('GET /posts'). Make sure it's inside describe('POST /posts/:id/like').

LemonPepperSeasoning committed:
reorganised the unit test into its own describe function

Dj5822 commented on backend/__tests__/posts.js:
```diff
@@ -402,4 +402,121 @@ describe('Posts', () => {
+
+                const authToken = Authentication.generateAuthToken(user1)
+
+                const createPostResponse = await request(server)
```
I made this same exact mistake, but we're not supposed to use other endpoints in the tests. What we have to do is to directly create a post by manipulating the model. You can check the updated version of my tests in my PR to get an idea of what I mean.

LemonPepperSeasoning replied:
Ok. 
Thanks for the great feedback !

Dj5822 reviewed (changes requested):
I realized that you made the same exact mistake that I did. It's not a major problem but it's probably best to fix it.

LemonPepperSeasoning committed:
Fixed unit test :
- my unit test will not use other endpoint to create model object.
- we directly create a model and only use endpoint that we are testing.

Dj5822 commented on backend/__tests__/posts.js:
```diff
@@ -402,4 +402,146 @@ describe('Posts', () => {
+                const invalidId = newPost.id + 999
+
+                const response = await request(server)
+                    .put(`/api/posts/${invalidId}/like`)
```
I think this part is supposed to be post instead of put.

LemonPepperSeasoning replied:
Yes. you are completely right!.
Sorry about that.

Dj5822 reviewed (changes requested):
One last thing.

LemonPepperSeasoning committed:
Fixed a typo in unit test

Dj5822 reviewed (commented):
LGTM

LemonPepperSeasoning committed:
Merge branch 'main' of https://github.com/SE701Team2/Updog into SE701Team2-main_2

LemonPepperSeasoning committed:
Merge branch 'SE701Team2-main_2' into like_post
